### PR TITLE
Strip @ character from branch name

### PR DESCRIPTION
**Pull request by conf**
Despite "@" character is perfectly fine as a part of a Git branch name,
it is not allowed inside a Docker tag. It can break some workflows when
branch name is used as a Docker tag for corresponding Docker image.

Replacing it with something safer like "at-" string solves the problem.
UPD: It's been decided to strip the `@` character from branch name altogether.

### CODE REVIEW

File: common/lib/dependabot/pull_request_creator/branch_namer.rb [hunk 1]
```diff
@@ -29,7 +29,8 @@ def new_branch_name
               elsif dependencies.count > 1 && updating_a_dependency_set?
                 dependency_set.fetch(:group)
               else
-                dependencies.map(&:name).join("-and-").tr(":", "-")
+                dependencies.map(&:name).join("-and-").tr(":", "-").
+                  gsub("@", "at-")
```
Comment (conf) [inline common/lib/dependabot/pull_request_creator/branch_namer.rb]: Moved to the next line because of rubocop 🤷‍♂

**Comment by conf** [comment]
Not sure why build fails for Python, failure seems unrelated to my changes. Could anybody have a look at this, please?

**Comment by conf** [comment]
@greysteil Please, have a look.

File: common/lib/dependabot/pull_request_creator/branch_namer.rb (hunk 1 above)
Comment (greysteil) [reply common/lib/dependabot/pull_request_creator/branch_namer.rb]: How about we just sub it out for nothing? I don't think we're likely to see collisions.

```suggestion
                  gsub("@", "")
```

**Comment by greysteil** [comment]
Thanks for this, and sorry for the delay reviewing it!

Can you please also sign this IP transfer letter? You can email the signed copy to support@dependabot.com.

(We're likely to change the license to a proper MSFT written one, or to MIT, in the future, and want make sure we own all the contributions to Dependabot so that we can do so. Sorry for the painful process!)

[Dependabot IP Letter - Alexey Shein.pdf](https://github.com/dependabot/dependabot-core/files/3342049/Dependabot.IP.Letter.-.Alexey.Shein.pdf)

**Comment by conf** [comment]
Thank you for looking into this! I'm fine with stripping `@` character altogether, I'll do the necessary changes shortly.
P.S. I'll send the signed IP transfer letter a bit later.

**Comment by greysteil** [comment]
PDF received - thanks! I'll get this deployed straight away.

**Comment by conf** [comment]
Thanks a lot! 🙏

**Pull request merged**